src/main/java/org/jenkinsci/plugins/workflow/graph/FlowNode.java

```diff
@@ -244,66 +248,123 @@ public String getUrl() throws IOException {
      *
      * This method provides such an opportunity for subtypes.
      */
-    protected void setActions(List<Action> actions) {
-        this.actions = new CopyOnWriteArrayList<Action>(actions);
+    protected synchronized void setActions(List<Action> actions) {
```

Comment:
Unnecessary to make this `synchronized` as it is only ever called from subclass constructors. (Really it should better have been made `final`: calling an overridable method from a constructor is inherently dangerous in Java, unlike in say C++.)